plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -425,12 +425,14 @@ public HiveTableHandle getTableHandle(ConnectorSession session, SchemaTableName
         if (isHiveSystemSchema(tableName.getSchemaName())) {
             return null;
         }
-        Optional<Table> table = metastore.getTable(new HiveIdentity(session), tableName.getSchemaName(), tableName.getTableName());
-        if (table.isEmpty()) {
-            return null;
-        }
+        return metastore.getTable(new HiveIdentity(session), tableName.getSchemaName(), tableName.getTableName())
+                .map(table -> getTableHandle(session, tableName, table))
+                .orElse(null);
+    }
 
-        if (isDeltaLakeTable(table.get())) {
+    private HiveTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName, Table table)
```

Comment:
I'm not sure if having a separate method is super useful here, we could store `Optional<Table>` as name optionalTable OR just do `metastore.getTable().orElse(null)`. but don't feel very strongly.

Reply:
You are right. The redirection in the code to another method is not necessary here.
I modified the code to have the logic of the `getTableHandle(...)` method contained where it was initially.